### ISSUE TITLE
Implement authToken

### ISSUE THREAD

**Issue by erickjx** [r-hannuschka/ngx-fileupload#258]
Can you implement  "Auth token" that will be applied as 'Authorization' header during file send, it used with JWT.

**Comment by r-hannuschka** [comment]
Thx for your feature request, i will try to implement this asap. This could may take 2 - 3 days since i have alot of work.

So for the first you can maybe make usage of an Interceptor to append your jwt.

Best regards,
Ralf

**Comment by r-hannuschka** [comment]
should be done, added some unit tests and checks if Access Token will be set. For this i updated NgxFileUploadOptions and added headers option to it, so you can define custom headers or a special authorization header which will automatically adds a 'Authorization' header to the request.

@example

```ts
@Component({
    selector: "app-ui--upload-toolbar",
    templateUrl: "upload-toolbar.component.html"
})
export class CustomUploadComponent {

    ...

    /**
      * drop files and create upload request
     */
    public drop(files: File[]) {
        const uploadOptions: NgxFileUploadOptions = {
                url: this.url, 
                headers: {
                     /** authorization header */
                     authorization: {
                         /**key: string by default bearer will send */
                         token: "My-Token"
                     },
                     /** custom header */
                     'X-RefKey': '0123456789abcdefgh'
                }
         };
        this.storage.add(
            this.uploadFactory.createUploadRequest(files, uploadOptions, this.validator));
    }
}
```

**Comment by r-hannuschka** [comment]
Just released 2.1.0 which implements headers configurations. If it not works let me know but i think it should.

Best regards,
Ralf

**Comment by r-hannuschka** [comment]
wait for response for at least 1 week

**Comment by r-hannuschka** [comment]
added docs for headers for example authorization header:

## Add Authorization Header

to provide send an authorization header we added **authorization** to headers

### Interface Authorization Header

```ts
interface AuthorizationHeader {
    key?: string;
    token: string;
}
```

### example

there are 3 ways to pass an authorization header

- passing a string value 

    ```ts
    /** upload options */
    const uploadOptions: UploadOptions = {
        url: "http://localhost:3000/upload/gallery",
        headers: {
            /**
             * will send header: 
             *
             * Authorization: Bearer my-token
             */
            authorization: "my-token",
        }
    };
    ```

- passing an object

    ```ts
    /** upload options */
    const uploadOptions: UploadOptions = {
        url: "http://localhost:3000/upload/gallery",
        headers: {
            /**
             * pass an object, if no key is passed it will take Bearer by default
             * so this is the same as authorization: "my-token"
             * 
             * Authorization: Bearer my-token
             */
            authorization: {token: "my-token" },
        }
    };
    ```

- passing an custom authorization header

    ```ts
    /** upload options */
    const uploadOptions: UploadOptions = {
        url: "http://localhost:3000/upload/gallery",
        headers: {
            /**
             * if you set an key this will be used as token name
             * sends header: 
             * 
             * Authorization: MyAuthToken my-token
             */
            authorization: {key: "MyAuthToken", token: "my-token" },
        }
    };
    ```
closed and implemented in 2.2.1
